server/channels/api4/channel.go

```diff
@@ -425,9 +425,18 @@ func restoreChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func createDirectChannel(c *Context, w http.ResponseWriter, r *http.Request) {
-	userIds := model.ArrayFromJSON(r.Body)
+	userIds, err := model.ArrayFromJSONNonSort(r.Body, *c.App.Config().ServiceSettings.MaximumPayloadSizeBytes)
+	if err != nil {
+		c.SetInvalidParam("user_ids")
```

Comment:
This actually gobbles up the returned error - making it to be no different than L441. Wondering if we can just convert the error to an AppError and set `c.Err = appError` and return. This way it's clear to the end user and also in the server logs as to what went wrong.

Reply:
Updated to return AppError in all cases.